graylog2-server/src/main/java/org/graylog2/decorators/LinkFieldConverter.java

```diff
@@ -0,0 +1,104 @@
+
+import com.floreysoft.jmte.Engine;
+import com.floreysoft.jmte.template.Template;
+import com.floreysoft.jmte.template.VariableDescription;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.assistedinject.Assisted;
+import org.graylog2.plugin.Message;
+import org.graylog2.plugin.configuration.ConfigurationRequest;
+import org.graylog2.plugin.configuration.fields.BooleanField;
+import org.graylog2.plugin.configuration.fields.TextField;
+import org.graylog2.plugin.decorators.SearchResponseDecorator;
+import org.graylog2.rest.models.messages.responses.ResultMessageSummary;
+import org.graylog2.rest.resources.search.responses.SearchResponse;
+
+import javax.inject.Inject;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+
+public class LinkFieldConverter implements SearchResponseDecorator {
+
+    private static final String CK_LINK_FIELD = "link_field";
+
+    private final String linkField;
+
+    public interface Factory extends SearchResponseDecorator.Factory {
+        @Override
+        LinkFieldConverter create(Decorator decorator);
+
+        @Override
+        LinkFieldConverter.Config getConfig();
+
+        @Override
+        LinkFieldConverter.Descriptor getDescriptor();
+    }
+
+    public static class Config implements SearchResponseDecorator.Config {
+
+        @Override
+        public ConfigurationRequest getRequestedConfiguration() {
+            return new ConfigurationRequest() {
+                {
+                    addField(new TextField(
+                            CK_LINK_FIELD,
+                            "Link field",
+                            "message",
+                            "The field that will be transformed into a hyperlink."
+                    ));
+                }
+            };
+        }
+    }
+
+    public static class Descriptor extends SearchResponseDecorator.Descriptor {
+        public Descriptor() {
+            super("Hyperlink String", "http://docs.graylog.org/", "Hyperlink string");
+        }
+    }
+
+    @Inject
+    public LinkFieldConverter(@Assisted Decorator decorator, Engine templateEngine) {
+        this.linkField = (String) requireNonNull(decorator.config().get(CK_LINK_FIELD),
+                                                   CK_LINK_FIELD + " cannot be null");
+    }
+
+    @Override
+    public SearchResponse apply(SearchResponse searchResponse) {
+        final List<ResultMessageSummary> summaries = searchResponse.messages().stream()
+                .map(summary -> {
+                    final Message message = new Message(ImmutableMap.copyOf(summary.message()));
+                    if (summary.message().containsKey(linkField)) {
+                        final String href = (String) summary.message().get(linkField);
+                        final LinkedHashMap object = new LinkedHashMap<String, String>();
```

Comment:
Here I see a couple of improvements:

- Use a regular `HashMap`, as we don't really care to maintain order in this case, and it's going to be a bit faster
- Use generics in the local variable you are storing the `Map`, avoiding unchecked calls to `put()` afterwards
- The name `object` is too generic for my taste, I would go with `decoratedField` or something similar

In summary, I would change this line into:
```
final Map<String, String> decoratedField = new HashMap<>();
```